components/extensions/org.wso2.carbon.ml.siddhi.extension/src/main/java/org/wso2/carbon/ml/siddhi/extension/PredictStreamProcessor.java

```diff
@@ -203,6 +274,12 @@ private void populateFeatureAttributeMapping() throws Exception {
         }
     }
 
+    private void logError(int modelId, Exception e) {
+        log.error("Error while retrieving ML-model : " + modelStorageLocations[modelId], e);
+        throw new ExecutionPlanCreationException(
```

Comment:
This is a CEP extension and this exception is thrown at the validation of the Siddhi query definition when creating an Execution plan. This exception is required to show what the error is, in execution plan creation UI.